src/main/java/com/kodilla/ecommercee/service/UserService.java

```diff
@@ -0,0 +1,63 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.exception.UserNotFoundException;
+import com.kodilla.ecommercee.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Random;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public List<User> getUsers() {
+        return userRepository.findAll();
+    }
+
+    public User getUser(Long id) throws UserNotFoundException {
```

Comment:
why do you include "throws UserNotFoundException" is it necessary ?

```suggestion
    public User getUser(Long id) {
```

Reply:
to be able to get HTTP response 404 instead of server internal error when I'm calling id of not existing user..

Reply:
RuntimeException is a unchecked exception as we mentioned on our meeting so we do not have to throw it.
I've check in postman w/o "throws UserNotFoundException" and i get 404, am I missing something?

Reply:
What does that have to do with whether it's checked or not? Unchecked exceptions don't have to be handled by the developer but in this case we want to handle it to achieve a specific result (according to the conversation in which you indicated that it is better to use an exception here, not an optional - we still want to keep appropriate HTTP response).

https://snipboard.io/XkQIGV.jpg - with exception
`public User getUser(Long id) throws UserNotFoundException {
        return userRepository.findById(id).orElseThrow(UserNotFoundException::new);
    }`

https://snipboard.io/wEdPg9.jpg - without exception
`public User getUser(Long id) {
       return userRepository.findById(id).get();
    }`

In one it returns 404 in the second 500. HTTP response is thrown now, after all, from the exception.
Please - verify again and let me know, because maybe we are talking about different classes/methods (here you pointed to UserService and getting a single user - getUser() .

Reply:
Ok - I see my mistake.